### PR TITLE
My project

### PR DESCRIPTION
**Pull request by Gheifr**
First home task reg personal project
*with questions

### CODE REVIEW

File: MyProject/cls/Employee.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+using System.Text;
+
+namespace cls
+{
+    class Employee: Person
+    {
+        public string Position { get; private set; }
+        public int AccessLevel { get; private set; }
+
+        private double Salary;
```
Comment (budda2) [inline MyProject/cls/Employee.cs]: private fields names should be in the form like : "_someNameField"

File: MyProject/cls/Employee.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+        public int userID { get; private set; }
+
+        public void SetSalary(double Salary)
```
Comment (budda2) [inline MyProject/cls/Employee.cs]: you have two methods named "SetSalary" and they do different stuff, please fix.

File: MyProject/cls/Guest.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace cls
+{
+    public class Guest:Person
+    {
+        public int Visits;
+        public string LoyaltyLevel;
```
Comment (budda2) [inline MyProject/cls/Guest.cs]: i would assume that this field should be reworked to property with privte set to make it impossible to change it from outside. Now it seems that LoyalityLevel could be set from anywhere

File: MyProject/cls/Person.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    public string Address{ get; private set; }
+    public string Email{ get; private set; }
+
+    private DateTime dateOfBirth;
+    public string Age { get; private set; }
+    
+        public string GetAge()
+        {
+            System.DateTime dateNow = DateTime.Now;
+            return ""+(dateNow.Year - dateOfBirth.Year);
```
Comment (budda2) [inline MyProject/cls/Person.cs]: you can call ToString() method on the result of your calculation. What you have done, will work, but that is nasty.

File: MyProject/cls/Restaurant.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace cls
+{
+       public class Restaurant
+    {
+        public string Name;
```
Comment (budda2) [inline MyProject/cls/Restaurant.cs:7]: as you have methods to set value of each of this fields, i guess what you need here is a properties with private set.

File: MyProject/cls/Balance.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace cls
+{
+    class Balance
+    {
+        //cashBalance
```
Comment (budda2) [inline MyProject/cls/Balance.cs:9]: uncommit this an fully implement

File: MyProject/cls/Employee.cs (hunk 2 above)
Comment (budda2) [inline MyProject/cls/Employee.cs]: parameter names should start with small letter: public void SomeMethod(Int someParameter){}

File: MyProject/cls/Payment.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace cls
+{
+    class Payment
+    {
+        //Cash
```
Comment (budda2) [inline MyProject/cls/Payment.cs:9]: Same as for the Balance

File: MyProject/cls/Table.cs [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+using System.Text;
+
+namespace cls
+{
+    public class Table
+    {
+        public int TblNumber { get; private set; }
+        public int Guests { get; private set; }
+
+        //_order[100] - question regarding whether to use array or collection?
```
Comment (budda2) [inline MyProject/cls/Table.cs:12]: for now go with Array, if you would like to use something else, go with HashSet<T>, read a documentation, understand what are the complexity of adding/getting items to/from this collection, what are the specific things about this sort of collection, and if you are fully with it, use it within your ptoject.

**Review by budda2 — COMMENTED** [review]
Code review done. Seems good to me. Overall progress is fine, when you are done with fixing my notes feel free to merge pull request by yourself.

File: MyProject/cls/Employee.cs (hunk 1 above)
Comment (Gheifr) [reply MyProject/cls/Employee.cs]: Done

File: MyProject/cls/Employee.cs (hunk 2 above)
Comment (Gheifr) [reply MyProject/cls/Employee.cs]: Done, one supposed to be Get instead of Set

File: MyProject/cls/Guest.cs (hunk 3 above)
Comment (Gheifr) [reply MyProject/cls/Guest.cs]: Done, added method to Get the level

File: MyProject/cls/Person.cs (hunk 4 above)
Comment (Gheifr) [reply MyProject/cls/Person.cs]: Fixed

File: MyProject/cls/Employee.cs (hunk 2 above)
Comment (Gheifr) [reply MyProject/cls/Employee.cs]: Done

File: MyProject/cls/Payment.cs (hunk 7 above)
Comment (Gheifr) [reply MyProject/cls/Payment.cs:9]: Left it, will need later.

**Pull request merged**